dist-git-to-src-git/dist2src.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/python3
+
+import os
+import tarfile
+import click
+import git
+import logging
+import sh
+from time import sleep
+
+
+logger = logging.getLogger(__name__)
+
+
+@click.group("dist2src")
+@click.option("-v", "--verbose", count=True)
+def cli(verbose):
+    logger.addHandler(logging.StreamHandler())
+    if verbose > 1:
+        logger.setLevel(logging.DEBUG)
+    elif verbose > 0:
+        logger.setLevel(logging.INFO)
+
+
+@cli.command()
+@click.argument("path")
+@click.argument("branch")
+@click.option("--orphan", is_flag=True, help="Create an branch with disconnected history.")
+def checkout(path, branch, orphan=False):
+    if path.startswith(os.pardir):
+        raise ValueError("This is bad, don't start path with '..'")
+
+    if not os.path.exists(path):
+        os.makedirs(path)
+
+    repo = git.Repo.init(path)
+    options = {}
+    if orphan:
+        options["orphan"] = branch
+    else:
+        options["B"] = branch
+
+    repo.git.checkout(**options)
+
+
+@cli.command()
+@click.argument("path")
+def get_archive(path):
+    # TODO(csomh): make the path to get_sources.sh configurable
+    get_sources = sh.Command("get_sources.sh")
```

Comment:
Where is the `get_sources.sh`? I don't see it in this PR.
Getting sources should be as easy as `spectool -g *.spec`.

Reply:
https://wiki.centos.org/Sources#get_sources.sh_script

Reply:
`spectool` downloads from the URL in the spec file. `get_sources.sh` downloads the archive from the lookaside cache as references in the [.metadata](https://git.centos.org/rpms/rpm/blob/c8s/f/.rpm.metadata) file.